docker/Dockerfile

```diff
@@ -1,25 +1,28 @@
-FROM ubuntu:xenial
+FROM debian:stretch-slim
 
-RUN apt-get update -y && apt-get upgrade -y && apt-get install -y \
-build-essential \
+RUN apt-get update -y && apt-get upgrade -y && apt-get install -y --no-install-recommends \
```

Comment:
is the `apt-get upgrade` necessary? will it install all the updates in the system?

Reply:
I guess removing it would be better, as we don't need to upgrade all the packages...

Reply:
In fact it seems to be [discouraged](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#run).